processing/src/main/java/org/apache/druid/segment/column/ColumnCapabilities.java

```diff
@@ -37,6 +37,12 @@
    */
   ValueType getType();
 
+  /**
+   *
+   * If ValueType is COMPLEX, then the typeName associated with it.
+   */
+  String getTypeName();
```

Comment:
could this be `getComplexTypeName` to tie it more strongly to complex types? It would also match the aggregator factory changes in #9638 which I've started working on again recently. If you recall this discussion https://github.com/apache/druid/pull/9638#discussion_r410995227, I decided to go ahead and rename it to `getComplexTypeName` there because there are already breaking changes with the new abstract methods, so might as well make it intuitive.

Reply:
sgtm

Reply:
changed